os/se/ameba/rtl_security_api_wrapper.c

```diff
@@ -0,0 +1,403 @@
+
+/* AES */
+static const unsigned char aes_iv[16] = {
+	0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+	0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F
+};
+
+
+/* Secure efuse key location */
+#define SAMSUNG_KEY_ADDR 0x150
+#define SAMSUNG_KEY_SIZE 32
+
+/* Secure Storage Variable */
+#define SEC_STORE_OFFSET 0x1C0000
+#define TAG_LENGTH 16
+
+static const unsigned char aes_test_iv_gcm[12] = {
+	0xca, 0xfe, 0xba, 0xbe, 0xfa, 0xce, 0xdb, 0xad,
+	0xde, 0xca, 0xf8, 0x88
+};
+
+static const unsigned char aes_test_aad_gcm[20] = {
+	0xfe, 0xed, 0xfa, 0xce, 0xde, 0xad, 0xbe, 0xef,
+	0xfe, 0xed, 0xfa, 0xce, 0xde, 0xad, 0xbe, 0xef,
+	0xab, 0xad, 0xda, 0xd2
+};
+
+/* Hal Data struct define */
+typedef struct _hal_data {
+	void *data;
+	uint32_t data_len;
+	void *priv;
+	uint32_t priv_len;
+} hal_data;
+
+static uint8_t samsung_key[32] = { 0 };
+static uint8_t crypto_inited = 0;
+
+/**
+ * Read secure efuse
+ */
+static void rtl_ReadEfuse(u32 addr, u8 size, u8* key)
+{
+	uint32_t Idx = 0;
+	uint32_t CtrlSetting = HAL_READ32(SYSTEM_CTRL_BASE, REG_HS_EFUSE_CTRL1_S);
+	
+	for (Idx = 0; Idx < size; Idx++) {
+		EFUSERead8(CtrlSetting, addr + Idx, &key[Idx], L25EOUTVOLTAGE);
+	}
+}
+
+/**
+ * Hardware Crpyto Wrapper
+ */
+uint32_t rtl_cryptoEngine_init_wrapper(void)
+{
+	uint32_t ret = RTL_HAL_SUCCESS;
+
+	if (!crypto_inited) {
+		device_mutex_lock(RT_DEV_LOCK_CRYPTO);
+		ret = rtl_cryptoEngine_init();	/* init crypto engine */
+		device_mutex_unlock(RT_DEV_LOCK_CRYPTO);
+
+		if (ret == RTL_HAL_SUCCESS) {
+			crypto_inited = 1;
+		} else {
+			return RTL_HAL_FAIL;
+		}
+	}
+
+	/* Read Samsung Key for Decyption */
+	device_mutex_lock(RT_DEV_LOCK_EFUSE);
+	rtl_ReadEfuse(SAMSUNG_KEY_ADDR, SAMSUNG_KEY_SIZE, samsung_key);
+	device_mutex_unlock(RT_DEV_LOCK_EFUSE);
+
+	return ret;
+}
+/**
+ * AES Crpyto Wrapper
+ */
+uint32_t rtl_cryptoAES_ecb_wrapper(uint8_t* key, uint32_t keylen, unsigned char* message, uint32_t msglen, unsigned char* pResult, uint8_t mode)
+{
+	uint32_t ret = RTL_HAL_SUCCESS;
+
+	device_mutex_lock(RT_DEV_LOCK_CRYPTO);
+	ret = rtl_crypto_aes_ecb_init(key, keylen);
+
+	/* Init Success */
+	if (ret == RTL_HAL_SUCCESS) {
+		if (mode == MBEDTLS_ENCRYPT) {
+			/* Encryption operation */
+			ret = rtl_crypto_aes_ecb_encrypt(message, msglen, NULL, 0, pResult);
+		} else if (mode == MBEDTLS_DECRYPT) {
+			/* Decryption operation */
+			ret = rtl_crypto_aes_ecb_decrypt(message, msglen, NULL, 0, pResult);
+		} else {
+			ret = RTL_HAL_NOT_SUPPORTED;
+		}
+	} else {
+		ret = RTL_HAL_FAIL;
+	}
+	device_mutex_unlock(RT_DEV_LOCK_CRYPTO);
+
+	return ret;
+}
+
+uint32_t rtl_cryptoAES_cbc_wrapper(uint8_t* key, uint32_t keylen, unsigned char* message, uint32_t msglen, unsigned char* pResult, uint8_t mode)
+{
+	uint32_t ret = RTL_HAL_SUCCESS;
+	unsigned char iv[16];
+
+	memcpy(iv, aes_iv, sizeof(aes_iv));
+	device_mutex_lock(RT_DEV_LOCK_CRYPTO);
+	ret = rtl_crypto_aes_cbc_init(key, keylen);
+
+	if (ret == RTL_HAL_SUCCESS) {
+		if (mode == MBEDTLS_ENCRYPT) {
```

Comment:
SE should not refer mbedtls definition(mbedtls_encrypt is defined at cipher.h)

Reply:
We have remove all mbedtls .h file, these are local define enum.
Which mean it does not refer to any application mbedtls.

Reply:
we have rename the define on latest commit